### PR TITLE
Add shield management for ElasticCloud

### PR DESCRIPTION
**Pull request by trossignol**
Thank you for proposing a pull request. This template will guide you through the essential steps necessary for a pull request.
Make sure that:
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [ ] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).
- [x] You provide your full name and an email address registered with your GitHub account. If you’re a first-time submitter, make sure you have completed the [Contributor’s License Agreement form](https://support.springsource.com/spring_committer_signup).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>org.springframework.data</groupId>
     <artifactId>spring-data-elasticsearch</artifactId>
-    <version>2.1.0.BUILD-SNAPSHOT</version>
+    <version>2.1.0.BUILD-TRO-SNAPSHOT</version>
```
Comment (martin-g) [inline pom.xml]: What is `TRO-`?
I guess it is your own custom version.

File: src/main/java/org/springframework/data/elasticsearch/client/TransportClientFactoryBean.java [hunk 2]
```diff
@@ -86,15 +88,20 @@ public void afterPropertiesSet() throws Exception {
 	}
 
 	protected void buildClient() throws Exception {
-		client = TransportClient.builder().settings(settings()).build();
+		client = TransportClient.builder().addPlugin(ShieldPlugin.class).settings(settings()).build();
```
Comment (martin-g) [inline src/main/java/org/springframework/data/elasticsearch/client/TransportClientFactoryBean.java]: IMO this should be optional.
E.g. if there is a bean of type `List<Class<elasticsearch.Plugin>>` in the app context than for each item in the list call `.addPlugin(item)`

File: pom.xml (hunk 1 above)
Comment (trossignol) [inline pom.xml]: Yes, it was a mistake.

**Comment by garpinc** [comment]
See #159 I think this should supersede this and hence this one should be closed.

**Comment by pivotal-issuemaster** [comment]
@trossignol Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=152) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Pull request closed without merging**